### PR TITLE
feat(vercel): allow a single Sentry project to map to multiple Vercel projects

### PR DESCRIPTION
**Pull request by scefali**
This PR allows a single Sentry project to map to multiple Vercel projects as some users have requested this. There was no real reason we prevented this before. Note that it's not possible to support multiple Sentry projects for a single Vercel project.

I also swapped the columns so the Vercel project is on the left which makes it less confusing when you are mapping to the same Sentry project:

Screen Shot 2022-03-16 at 1 52 30 PM

### CODE REVIEW

**Comment by ceorourke** [comment]
Can you add a test for this case?

**Comment by scefali** [comment]
@ceorourke tests added

**Comment by ceorourke** [comment]
It's been a while since I've worked on Vercel so bear with me: what does this change look like in the UI? Right now we have a 1:1 Sentry to Vercel project mapping, if you are able to map 1 to many then what would that look like? Would it be clear which projects it's mapping to?

**Comment by scefali** [comment]
> It's been a while since I've worked on Vercel so bear with me: what does this change look like in the UI? Right now we have a 1:1 Sentry to Vercel project mapping, if you are able to map 1 to many then what would that look like? Would it be clear which projects it's mapping to?

@ceorourke Here's what it looks like:

Screen Shot 2022-03-16 at 10 51 08 AM

File: tests/js/spec/components/deprecatedforms/projectMapperField.spec.jsx [hunk 1]
```diff
@@ -72,5 +72,22 @@ describe('ProjectMapperField', () => {
...
+    // but not the value
+    openMenu(wrapper, {control: true, name: 'mappedDropdown'});
+    expect(
+      findOption(wrapper, {value: '1'}, {control: true, name: 'mappedDropdown'})
+    ).toHaveLength(0);
+    // validate we can still find 2
+    expect(
+      findOption(wrapper, {value: '2'}, {control: true, name: 'mappedDropdown'})
+    ).toHaveLength(1);
+  });
```
Comment (ceorourke) [inline tests/js/spec/components/deprecatedforms/projectMapperField.spec.jsx:92]: In this test I'm a little confused what 24, 1, and 2 are. 24 is 'project' and 1 and 2 are both 'mappedDropdown' - is 24 the Sentry project and 1 and 2 are the Vercel projects, or the other way around?

Comment (scefali) [reply tests/js/spec/components/deprecatedforms/projectMapperField.spec.jsx:92]: @ceorourke 23/24 are the Sentry projects and 1/2 are the Vercel projects (aka the mapped value)

**Review by ceorourke — APPROVED** [review]

**Pull request merged**